subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -250,41 +246,84 @@ static int adv_send(struct bt_mesh_ext_adv *ext_adv, struct bt_mesh_adv *adv)
 	return err;
 }
 
-static const char * const adv_tag_to_str[] = {
-	[BT_MESH_ADV_TAG_LOCAL]  = "local adv",
-	[BT_MESH_ADV_TAG_RELAY]  = "relay adv",
-	[BT_MESH_ADV_TAG_PROXY]  = "proxy adv",
-	[BT_MESH_ADV_TAG_FRIEND] = "friend adv",
-	[BT_MESH_ADV_TAG_PROV]   = "prov adv",
-};
-
-static bool schedule_send_with_mask(struct bt_mesh_ext_adv *ext_adv, int ignore_mask)
+static bool stop_proxy_adv(struct bt_mesh_ext_adv *ext_adv)
 {
+	int err;
+
 	if (atomic_test_and_clear_bit(ext_adv->flags, ADV_FLAG_PROXY)) {
-		atomic_clear_bit(ext_adv->flags, ADV_FLAG_PROXY_START);
-		(void)bt_le_ext_adv_stop(ext_adv->instance);
+		err = bt_le_ext_adv_stop(ext_adv->instance);
+		if (err) {
+			LOG_ERR("Failed to stop proxy advertising: %d", err);
+			atomic_set_bit(ext_adv->flags, ADV_FLAG_PROXY);
+			return false;
+		}
 
 		atomic_clear_bit(ext_adv->flags, ADV_FLAG_ACTIVE);
 	}
 
+	return true;
+}
+
+static bool schedule_send_with_mask(struct bt_mesh_ext_adv *ext_adv, int ignore_mask)
```

Comment:
This function is only called once now from `schedule_send` and `ignore_mask` is always 0.

Reply:
yeah, actually after remove while 1 in workq thread, `schedule_send_with_mask` will used.